paraview/xmls/CompactTriangulationPreconditioning.xml

```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<ServerManagerConfiguration>
+  <ProxyGroup name="filters">
+    <SourceProxy name="ttkCompactTriangulationPreconditioning" class="ttkCompactTriangulationPreconditioning" label="TTK CompactTriangulationPreconditioning">
+       <Documentation long_help="TTK plugin for the generation of clusters before using TopoCluster." short_help="TTK preprocessing plugin for TopoCluster.">
+        Given a simplicial mesh, this filter uses the PR star octree to divide 
+        the mesh into different regions, and adds this clustering information as 
+        a new scalar field to the original dataset. This clustering index scalar 
+        field can be further used by TopoCluster data structure.
+
+        Related publications:
+        "The PR-star octree: A spatio-topological data structure for tetrahedral meshes." 
+        Kenneth Weiss, Leila Floriani, Riccardo Fellegara, and Marcelo Velloso
+        Proc. of ACM SIGSPATIAL 2011.
+        
+        "TopoCluster: A Localized Data Structure for Topology-based Visualization"
+        Guoxi Liu, Federico Iuricich, Riccardo Fellegara, and Leila De Floriani
+        IEEE Transactions on Visualization and Computer Graphics, 2021.
+       </Documentation>
+
+      <!-- INPUT -->
+         <InputProperty name="Input" command="SetInputConnection">
+          <ProxyGroupDomain name="groups">
+            <Group name="sources" />
+            <Group name="filters" />
+          </ProxyGroupDomain>
+          <DataTypeDomain name="input_type">
+            <DataType value="vtkPointSet" />
+          </DataTypeDomain>
+          <!-- <InputArrayDomain name="input_array" attribute_type="point"
+                number_of_components="1" />
+          <Documentation>A vtkDataSet that has at least one point data scalar array that can be used to compute averages.</Documentation> -->
+        </InputProperty>
+
+      <!-- INPUT PARAMETER WIDGETS -->
+        <StringVectorProperty name="SelectedFields" command="SetScalarField" element_types="2 0" information_property="CellArrayInfo" label="Fields" number_of_elements="2" number_of_elements_per_command="1" repeat_command="1">
```

Comment:
what is the purpose of this field selection?
as far as I understand, this filter only processes a triangulation, irrespective of the data defined on it.

thanks for your feedback.

Reply:
btw, in paraview's gui, turning them off (by unchecking the corresponding checkbox), applying, then turning them on again, then applying, results in a crash.

Reply:
> what is the purpose of this field selection? as far as I understand, this filter only processes a triangulation, irrespective of the data defined on it.
> 
> thanks for your feedback.

The purpose here is to get the original fields in the data set and modify the values since the vertex indices could get changed by the clustering method. 


> btw, in paraview's gui, turning them off (by unchecking the corresponding checkbox), applying, then turning them on again, then applying, results in a crash.

I will look into the issue and solve it.

Reply:
I see, thanks for the feedback.
if you reproduce (and fix) this issue, that'd be fantastic.

Reply:
one quick remark @guoxiliu:
wouldn't it make more sense, instead of shuffling all data arrays, to generate only one new data array storing the correspondence of indices between new and old identifiers?
(for instance called `ttkCompactTriangulationIdentifiers`).

if the input contains a large number of data arrays, only one new data array would need to be produced without shuffling the entire set of input data arrays.

I am also concerned about possible inconsistencies, with regard to the VTK pipeline further down that filter, resulting from swapping data values.

finally, at a conceptual level, again, this filter should not modify the data attached to the input geometry as its purpose is only to provide an alternative geometry representation. in principle, that filter could be applied without any data defined on the geometry. that would argue overall for storing identifier correspondences in an extra array instead of shuffling data arrays.

what do you think?

Reply:
Hi @julien-tierny,

I see your reason here for not modifying the original data. The only concern for me with the additional array is how the data array is going to be accessed later (as you mentioned). I don't see an easy way to do that. 

Right now the user can select which data arrays for preconditioning so that the plugin doesn't shuffle all the input data arrays. Though I was still fixing the issue, maybe it's a workaround?

Reply:
I experimented a bit with the filter and I agree that there is no easy workaround this unfortunately.
thanks for the info!